Problems with installing mgra

Hello, 

I am trying to compile and install mgra on my system. Current system is running on ubuntu 20.04.2, with cmake 3.16.3 and gcc 9.3.0. cmake generates the makefile, but the make command fails to execute with errors. The copy of errors is attached below for your reference. Thanks in advance! 

make
[ 58%] Built target BLOSSOM5
[ 64%] Building CXX object src/utils/io/CMakeFiles/input.dir/path_helper.cpp.o
In file included from /home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:28,
                 from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/detail/shared_count.hpp:395:33: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  395 |     explicit shared_count( std::auto_ptr<Y> & r ): pi_( new sp_counted_impl_p<Y>( r.get() ) )
      |                                 ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:242:65: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  242 | template< class T, class R > struct sp_enable_if_auto_ptr< std::auto_ptr< T >, R >
      |                                                                 ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:441:31: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  441 |     explicit shared_ptr( std::auto_ptr<Y> & r ): px(r.get()), pn()
      |                               ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:454:22: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  454 |     shared_ptr( std::auto_ptr<Y> && r ): px(r.get()), pn()
      |                      ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:520:34: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  520 |     shared_ptr & operator=( std::auto_ptr<Y> & r )
      |                                  ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:529:34: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  529 |     shared_ptr & operator=( std::auto_ptr<Y> && r )
      |                                  ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/shared_ptr.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/format/alt_sstream.hpp:21,
                 from /home/mom/mgra-master/src/libs/boost/format/internals.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:38,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp: In member function ‘boost::shared_ptr<T>& boost::shared_ptr<T>::operator=(std::auto_ptr<_Up>&&)’:
/home/mom/mgra-master/src/libs/boost/smart_ptr/shared_ptr.hpp:531:38: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
  531 |         this_type( static_cast< std::auto_ptr<Y> && >( r ) ).swap( *this );
      |                                      ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
In file included from /home/mom/mgra-master/src/libs/boost/mpl/aux_/na_assert.hpp:23,
                 from /home/mom/mgra-master/src/libs/boost/mpl/arg.hpp:25,
                 from /home/mom/mgra-master/src/libs/boost/mpl/placeholders.hpp:24,
                 from /home/mom/mgra-master/src/libs/boost/iterator/iterator_categories.hpp:17,
                 from /home/mom/mgra-master/src/libs/boost/iterator/iterator_facade.hpp:14,
                 from /home/mom/mgra-master/src/libs/boost/range/iterator_range_core.hpp:27,
                 from /home/mom/mgra-master/src/libs/boost/range/iterator_range.hpp:13,
                 from /home/mom/mgra-master/src/libs/boost/range/as_literal.hpp:22,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string/trim.hpp:19,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string.hpp:19,
                 from /home/mom/mgra-master/src/utils/logger/logger_impl.hpp:20,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:145,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/mpl/assert.hpp: At global scope:
/home/mom/mgra-master/src/libs/boost/mpl/assert.hpp:188:21: error: unnecessary parentheses in declaration of ‘assert_arg’ [-Werror=parentheses]
  188 | failed ************ (Pred::************
      |                     ^
/home/mom/mgra-master/src/libs/boost/mpl/assert.hpp:193:21: error: unnecessary parentheses in declaration of ‘assert_not_arg’ [-Werror=parentheses]
  193 | failed ************ (boost::mpl::not_<Pred>::************
      |                     ^
In file included from /home/mom/mgra-master/src/libs/boost/bind/mem_fn.hpp:25,
                 from /home/mom/mgra-master/src/libs/boost/mem_fn.hpp:22,
                 from /home/mom/mgra-master/src/libs/boost/function/detail/prologue.hpp:18,
                 from /home/mom/mgra-master/src/libs/boost/function.hpp:24,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string/detail/find_iterator.hpp:18,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string/find_iterator.hpp:24,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string/iter_find.hpp:27,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string/split.hpp:16,
                 from /home/mom/mgra-master/src/libs/boost/algorithm/string.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/logger_impl.hpp:20,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:145,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/home/mom/mgra-master/src/libs/boost/get_pointer.hpp:27:40: error: ‘template<class> class std::auto_ptr’ is deprecated [-Werror=deprecated-declarations]
   27 | template<class T> T * get_pointer(std::auto_ptr<T> const& p)
      |                                        ^~~~~~~~
In file included from /usr/include/c++/9/bits/locale_conv.h:41,
                 from /usr/include/c++/9/locale:43,
                 from /home/mom/mgra-master/src/libs/boost/format.hpp:23,
                 from /home/mom/mgra-master/src/utils/logger/perfcounter.hpp:10,
                 from /home/mom/mgra-master/src/utils/logger/logger.hpp:8,
                 from /home/mom/mgra-master/src/utils/io/path_helper.hpp:16,
                 from /home/mom/mgra-master/src/utils/io/path_helper.cpp:7:
/usr/include/c++/9/bits/unique_ptr.h:53:28: note: declared here
   53 |   template<typename> class auto_ptr;
      |                            ^~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/utils/io/CMakeFiles/input.dir/build.make:63: src/utils/io/CMakeFiles/input.dir/path_helper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:201: src/utils/io/CMakeFiles/input.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,7 +37,7 @@ set(ROOT_DIR "${CMAKE_SOURCE_DIR}/..")
 set(COMPILE_OPTION_FLAG "-std=c++11" )
 set(DEBUG_FLAG "-g3")
 set(OPTIMIZE_FLAG "-O2 -pipe -finline-functions -finline-limit=1000") # -DPSEUDO_EDGE
-set(WARNS "-Wall -Werror -Wextra -pedantic-errors")
+set(WARNS "-Wall -Wextra -pedantic-errors")
 
 if (CMAKE_BUILD_TYPE STREQUAL "Release")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_OPTION_FLAG} ${OPTIMIZE_FLAG} ${WARNS}")

